product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/vfs/VFSClient.java

```diff
@@ -0,0 +1,222 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.scenario.test.common.vfs;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.net.ftp.FTP;
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPFile;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * VFS client for vfs operations
+ */
+public class VFSClient {
+    private String hostname;
+    private String userName;
+    private String password;
+    private int port;
+    private FTPClient ftpClient;
+
+    private static final Log log = LogFactory.getLog(VFSClient.class);
+
+    /**
+     * Constructor to VFSClient
+     *
+     * @param hostname host name or IP address of the FTP server
+     * @param username username of an FTP account on the FTP server to connect to
+     * @param password password corresponds to the username
+     * @param port ftp port
+     */
+    public VFSClient(String hostname, String username, String password, int port) {
+        this.hostname = hostname;
+        this.userName = username;
+        this.password = password;
+        this.port = port;
+        this.ftpClient = new FTPClient();
+    }
+
+    /**
+     * Function to connect to an FTP Server
+     *
+     * @throws IOException if connecting to FTP server fails
+     */
+    public void connectToFtp() throws IOException {
+        try {
+            ftpClient.connect(hostname, port);
+            ftpClient.login(userName, password);
+            ftpClient.enterLocalPassiveMode();
+        } catch (IOException ex) {
+            throw new IOException("IO Exception occurred while connecting to FTP server", ex);
+        }
+    }
+
+    /**
+     * Function to upload a file to an FTP Server
+     *
+     * @param sourceFilePath source file path
+     * @param targetFilePath target path of the remote file
+     * @throws IOException if file upload fails
+     */
+    public void ftpFileUpload(String sourceFilePath, String targetFilePath) throws IOException {
+        ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+        File localFile = new File(sourceFilePath);
+
+        try (InputStream inputStream = new FileInputStream(localFile)) {
+            log.info("Start uploading file " + sourceFilePath + " to " + targetFilePath);
+
+            try {
+                if (ftpClient.storeFile(targetFilePath, inputStream)) {
+                    log.info("File has been uploaded to " + targetFilePath + " successfully.");
+                }
+            } catch (IOException ex) {
+                throw new IOException("IO Exception occurred while uploading file ", ex);
```

Comment:
Do we really need to catch the exception and create a new exception? The method call will anyway be there in the stacktrace.

Reply:
IO Exception would be thrown when downloading the file fails or unable to find the file. So isn't it better to throw the exception with a descriptive message?